master/src/main/java/org/apache/celeborn/service/deploy/master/clustermeta/AbstractMetaManager.java

```diff
@@ -330,8 +338,10 @@ public void restoreMetaFromFile(File file) throws IOException {
               .peek(
                   workerInfo -> {
                     // Reset worker's network location with current master's configuration.
-                    workerInfo.networkLocation_$eq(
-                        resolveMap.get(workerInfo.host()).get().getNetworkLocation());
+                    if (workerInfo.networkLocation().equals(NetworkTopology.DEFAULT_RACK)) {
+                      workerInfo.networkLocation_$eq(
+                          resolveMap.get(workerInfo.host()).get().getNetworkLocation());
```

Comment:
I don't see a change in `PbSerDeUtils.toPbWorkerInfo()` so that networkLocation is also persisted in the file. So, incase the worker is sending its location, it will not be recovered from Ratis. 
Also, if we persist it to file, then we don't need to 332-335 right?

Reply:
`PbSerDeUtils.fromPbWorkerResource` and `PbSerDeUtils.toPbWorkerResource` already have `networkLocation as a field. We persist it to Ratis in `AbstractMetaManager.updateRegisterWorkerMeta`. We need the L332-L335 in case the worker's networkLocation is still set to default rack. In this case, we can retry to see if it can be resolved.

Reply:
@akpatnam25, if the network location is null or empty, do we need to reset worker's network location?

Reply:
I don't think `workerInfo.networkInfo()` would ever be null since it is set to DEFAULT_RACK already: https://github.com/akpatnam25/incubator-celeborn/blob/CELEBORN-1313/common/src/main/scala/org/apache/celeborn/common/meta/WorkerInfo.scala#L45

Reply:
Let me know if I am missing something though